File: src/Emergence.Data.Shared/Models/InventoryItem.cs
using System;
using System.Text.Json.Serialization;
using Emergence.Data.Shared.Enums;

namespace Emergence.Data.Shared.Models
{
    public class InventoryItem
    {
        public int InventoryItemId { get; set; }
        [JsonConverter(typeof(EnumDisplayConverter<ItemType>))]
        public ItemType ItemType { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        [JsonConverter(typeof(EnumDisplayConverter<ItemStatus>))]
        public ItemStatus Status { get; set; }
        public DateTime? DateAcquired { get; set; }
        [JsonConverter(typeof(EnumDisplayConverter<Visibility>))]
        public Visibility Visibility { get; set; }
        public int? UserId { get; set; }
        public string CreatedBy { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime? DateCreated { get; set; }
        public DateTime? DateModified { get; set; }

        public Origin Origin { get; set; }
        public Inventory Inventory { get; set; }
        public UserSummary User { get; set; }
    }
}

File: src/Emergence.Web/ClientApp/src/app/shared/models/inventory-item.ts
import { User } from "./user";
import { Origin } from "./origin";
import { Inventory } from "./inventory";
import { ItemType } from "./enums";

export interface InventoryItem {
  inventoryItemId: number;
  itemType: ItemType;
  name: string;
  quantity: number;
  status: number;
  dateAcquired: Date;
  visibility: number;
  userId: number;
  createdBy: string;
  dateCreated: Date;
  modifiedBy: string;
  dateModified: Date;
  inventory: Inventory;
  origin: Origin;
  user: User;
}
